GrappleGirlGame/main.lua

```diff
@@ -1,66 +1,100 @@
-FLOOR_CATAGORY = 1
+FLOOR_CATEGORY = 4
 
 function love.load()
     baseWorld = love.physics.newWorld(0, 1000, false)
-    gGirl = Character:new(nil, baseWorld, {love.graphics.getWidth()/2,100},{400, 400});
-
+    gGirl = Character:new(nil, baseWorld, {love.graphics.getWidth() / 2, 100}, {400, 400})
 
     -- floor --
     floor = {}
     floor.shape = love.physics.newRectangleShape(love.graphics.getWidth(), 100)
-    floor.body = love.physics.newBody( baseWorld, love.graphics.getWidth()/2, love.graphics.getHeight(), "static")
+    floor.body = love.physics.newBody(baseWorld, love.graphics.getWidth() / 2, love.graphics.getHeight(), "static")
     floor.fixture = love.physics.newFixture(floor.body, floor.shape, 1)
-    floor.fixture:setFriction(1);
-    floor.fixture:setCategory(FLOOR_CATAGORY);
+    floor.fixture:setFriction(1)
+    floor.fixture:setCategory(FLOOR_CATEGORY, 10, 9)
     -- floor end --
 
-    -- grappleAnchor --
-    grappleAnchor = {}
-    grappleAnchor.shape = love.physics.newRectangleShape(50, 50)
-    grappleAnchor.body = love.physics.newBody( baseWorld, love.graphics.getWidth()/2, love.graphics.getHeight()/2, "static")
-    grappleAnchor.fixture = love.physics.newFixture(grappleAnchor.body, grappleAnchor.shape, 1)
-    grappleAnchor.fixture:setFriction(1);
-    grappleAnchor.fixture:setCategory(FLOOR_CATAGORY);
-    -- grappleAnchor end --
+    -- grappleAnchorBlock --
+    grappleAnchorBlock = {}
+    grappleAnchorBlock.shape = love.physics.newRectangleShape(50, 50)
+    grappleAnchorBlock.body =
+        love.physics.newBody(baseWorld, love.graphics.getWidth() / 2, love.graphics.getHeight() / 2, "kinematic")
+    grappleAnchorBlock.fixture = love.physics.newFixture(grappleAnchorBlock.body, grappleAnchorBlock.shape, 1)
+    grappleAnchorBlock.fixture:setFriction(1)
+    grappleAnchorBlock.fixture:setCategory(FLOOR_CATEGORY)
+    -- grappleAnchorBlock end --
+
+    baseWorld:setCallbacks(baseWorld.beginContact, baseWorld.endContact, mypresolve, mypostSolve)
+end
 
+function doesContainCatagory(fixt, cat)
+    local cats = {fixt:getCategory()}
+    for i = 1, #cats do
+        if cats[i] == cat then
+            return true
+        elseif cats[i] > cat then
+            return false
+        end
+    end
+    return false
 end
 
 function love.update(dt)
-    local contacts = baseWorld:getContacts( )
-    for i = 0,#baseWorld:getContacts( ) do
-        if contacts[i] ~= nil then
-            f1, f2 = contacts[i]:getFixtures()
-            if f1:getCategory() == FLOOR_CATAGORY and f2:getCategory() == CHARACTER_CATAGORY or f2:getCategory() == FLOOR_CATAGORY and f1:getCategory() == CHARACTER_CATAGORY then
-                if f1:getCategory() == FLOOR_CATAGORY then
-                    charFic=f2
-                else
-                    charFic=f1
-                end
-                charFic:getUserData().canJump = true
+    local contacts = baseWorld:getContacts()
+
+    for i = 1, #baseWorld:getContacts() do
```

Comment:
Try to add more comments here, this is a very large chunk of uncommented code.

Reply:
lol, the whole thing is a large chunk of uncommented code